.pre-commit-config.yaml

```diff
@@ -0,0 +1,33 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/timothycrosley/isort
+    rev: 5.6.4
+    hooks:
+      - id: isort
+        files: .*.py
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: ["--line-length", "79"]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: check-ast
+      - id: check-case-conflict
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: debug-statements
+      - id: check-added-large-files
+      - id: check-docstring-first
+      - id: check-yaml
+        exclude: conda/meta.yaml
+  - repo: https://github.com/PyCQA/flake8
+    rev: 3.8.4
+    hooks:
+      - id: flake8
```

Comment:
No need for `flake8` if using `black`.

Reply:
These two tools achieve different purposes. Black is a code formatter, flake8 is a compliance checker. So the two together do make sense.

Reply:
I think because `black` runs first `flake8` will only ever see blackened code?

Reply:
The hooks are run in the order of declaration, so yes, the goal is that flake8 analyses only the code once cleaned.

You are correct, `black` does just formatting following the rules that you asked for. It does not do any analysis. There's a `black` plugin for flake8 but I have not used it yet.